applications/web/templates/ingress.yaml

```diff
@@ -59,38 +64,37 @@ spec:
   {{- end }}
 
   rules:
-    {{- range .Values.ingress.hosts }}
-      - host: {{ . }}
-        http:
-          paths:
-            {{ if gt (len $customPaths) 0 }}
-            {{- range $customPaths }}
-            - path: {{ .path }}
-              backend:
-                serviceName: {{ .serviceName }}
-                servicePort: {{ .servicePort }}
-            {{- end }}
-            {{ else }}
-            - backend:
-                serviceName: {{ $fullName }}
-                servicePort: {{ $svcPort }}
-            {{ end }}
-    {{- end }}
-    {{- range .Values.ingress.porter_hosts }}
+    {{- range $allHosts }}
       - host: {{ . }}
         http:
           paths:
             {{ if gt (len $customPaths) 0 }}
             {{- range $customPaths }}
             - path: {{ .path }}
+              pathType: Prefix
               backend:
-                serviceName: {{ .serviceName }}
+                {{- if $.Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}                
+                service:
+                  name: {{ .serviceName }}
+                  port:
+                    number: {{ .servicePort }}
+                {{- else }}
+                serviceName: {{ .serviceName }} 
                 servicePort: {{ .servicePort }}
+                {{- end }}
             {{- end }}
             {{ else }}
             - backend:
```

Comment:
```suggestion
```

I think this lingering `backend:` block is throwing things off. Here's the block that worked for me:

```
            - {{- if $.Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
              pathType: ImplementationSpecific
              {{- end }}
              backend:
              {{- if $.Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
                service:
                  name: {{ $fullName }}
                  port:
                    number: {{ $svcPort }}
              {{- else }}
                serviceName: {{ $fullName }}
                servicePort: {{ $svcPort }}
              {{- end }}
```